e2e/webServer.sh

```diff
@@ -1,7 +1,38 @@
 #!/bin/bash
 
-# runs from e2e folder
+# export the following before running tests
+# export MONGO_INITDB_ROOT_USERNAME=somebody
+# export MONGO_INITDB_ROOT_PASSWORD=something
+#
+# This connection string is for IQEngine to use for connecting to
+# the database. It's connecting from its own container to Mongo
+# in its container. Hence, it uses iqenginedb:27017 as the url.
+#
+# Test for the name/pw
+if [[ -z "$MONGO_INITDB_ROOT_USERNAME" ]]; then
+  echo "MONGO_INITDB_ROOT_USERNAME must be set."
+  exit
+fi
+if [[ -z "$MONGO_INITDB_ROOT_PASSWORD" ]]; then
+  echo "MONGO_INITDB_ROOT_PASSWORD must be set."
+  exit
+fi
+
+# runs from e2e folder because that's where the playwright config files are
 cd .. || exit
+conn="mongodb://$MONGO_INITDB_ROOT_USERNAME:$MONGO_INITDB_ROOT_PASSWORD@iqenginedb:27017/admin"
 docker build -t iqengine .
-docker run -p 3000:3000 -e IN_MEMORY_DB=0 -e IQENGINE_FEATURE_FLAGS='{"useAPIDatasources": true}' iqengine:latest
+docker network create iqenginenet
+docker run --network iqenginenet -p 27017:27017 -d \
+  -e MONGO_INITDB_ROOT_USERNAME="$MONGO_INITDB_ROOT_USERNAME" \
+  -e MONGO_INITDB_ROOT_PASSWORD="$MONGO_INITDB_ROOT_PASSWORD" \
+  --name iqenginedb \
+  mongo:latest
+docker run --network iqenginenet -p 3000:3000 \
+  -e IN_MEMORY_DB=0 \
+  -e VITE_FEATURE_FLAGS='{"useAPIDatasources": true}' \
+  -e IQENGINE_METADATA_DB_CONNECTION_STRING="$conn" \
+  --name iqengine \
+  iqengine:latest
```

Comment:
This would probably be a docker compose maybe use the same docker compose that spin up the plugins.

Reply:
can't use the same compose yml file. Need one that uses newly built image rather than the one on ghcr.io